MED-02

ryszardpanda opened the pull request:
+ CRUD operations for patient

ryszardpanda committed:
MED-02
+ CRUD operations for patient

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/LombokPatient.java:
```diff
@@ -0,0 +1,22 @@
+
+@Data
+//@Builder
+public class LombokPatient {
```
Po prostu Patient a tamten Patient usuńmy

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/LombokPatient.java:
```diff
@@ -0,0 +1,22 @@
+//@ToString
+//@EqualsAndHashCode
+@AllArgsConstructor
+
```
pusta linia zbędna

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/MedicalClinicApplication.java:
```diff
@@ -0,0 +1,13 @@
+@SpringBootApplication
+public class MedicalClinicApplication {
+
+	public static void main(String[] args) {
```
ctrl alt l

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/PatientController.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @PostMapping("/patients")
+    public List<Patient> addPatients(@RequestBody List<Patient> patients) {
```
Nie było takiej funkcjonalności

konradgab replied:
Dodanie pojedynczego pacjenta

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/PatientController.java:
```diff
@@ -0,0 +1,55 @@
+    // Metoda DELETE, która usunie pacjenta na podstawie adresu e-mail
+    //nie ogarnalbym
+    @DeleteMapping("/patients/{email}")
+    public ResponseEntity<Void> deletePatientByEmail(@PathVariable String email) {
```
ResponseEntity to klasa reprezentująca odpowiedź HTTP. W sytuacji jak używasz adnotacji @RestController to automatycznie masz zamiane tego co zwracasz czyli np tak jak w powyzszym endpoincie zwracasz Patient to ten Patient trafia do body a Spring automatycznie obudowuje to body (Patient) w odpowiedź HTTP z statusem sukcesu 200 OK jeśli zapytanie się powiedzie. Defaultowy mechanizm Springa w sytuacji kiedy poleci w ramach przetwarzania zapytania wyjątek automatycznie zmapuje to na odpowiedź HTTP z kodem 500 INTERNAL_SERVER_ERROR.

konradgab replied:
Na ten moment zrób po prostu void. Taka metoda nie będzie zwracała nic w body a więc mamy status HTTP 204 NO_CONTENT. Jeśli chcemy aby taki response został nadany do odpowiedzi, które udało się przetworzyć wystarczy nad metodą w kontrolerze dodać adnotację @ResponseStatus(HTTPStatus.NO_CONTENT)

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/PatientController.java:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+
+
```
nadmiarowa pusta linia

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/PatientRepository.java:
```diff
@@ -0,0 +1,48 @@
+@RequiredArgsConstructor
+public class PatientRepository {
+
+    private final List<Patient> patients;
```
To co robi Spring w sytuacji jak wstrzykujesz Listę czegokolwiek to on wtedy przeszukuje wszystkie zdefiniowane beany dla danego typu i wstrzykuje wsystkie znalezione beany. Przykładowo jest to przydatne w sytuacji jak mamy np. jeden interfejs i 5 jego implementacji.

konradgab replied:
W twojej sytuacji nie znajdzie beana dla Paitentów wiec lista bedzie pusta na poczatku.

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/PatientRepository.java:
```diff
@@ -0,0 +1,48 @@
+        return patient;
+    }
+
+    public Patient getPatientByEmail(String email) {
```
Czy kojarzymy Optionala?

konradgab replied:
Optionala używamy aby nie zwracac nullow -> operacje na nullu powodują rzucenie NPE. Wszedzie tam gdzie metoda nie ma pewnosci ze cos istnieje a ma cos zwrocic powinna zwracac Optional -> Optional (Patient) wtedy swiadomie wskauzje ze osoba uzywajaca tej metody musi obsluzyc sytuacje kiedy bede mial nulla

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/PatientRepository.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public Patient updatePatient(String email, Patient updatedPatient) {
+        for (Patient patient : patients) {
```
Optional(Patient) optionalPatient = getPatientByEmail(email);
optionalPatient .ifPresent(patient -> {
                patient.setName(updatedPatient.getName());
                patient.setSurname(updatedPatient.getSurname());
                patient.setEmail(updatedPatient.getEmail());
}
return optionalPatient;
)

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/PatientService.java:
```diff
@@ -0,0 +1,37 @@
+        return patientRepository.getPatients();
+    }
+
+    public List<Patient> addPatients(List<Patient> patients) {
```
powinno dodać pojedynczego pacjenta

konradgab commented on src/main/java/com/ryszardpanda/medicalClinic/Patient.java:
```diff
@@ -0,0 +1,12 @@
+
+@AllArgsConstructor /// chce miec konstruktor do wsystkich argumentow
+@Data // chce uzywac daty
+public class Patient {
```
Do usuniecia

ryszardpanda committed:
MED-02
+ FIX CRUD operations for patient from Konrad comments

ryszardpanda committed:
MED-02
+ FIX CRUD operations for patient from Konrad comments

ryszardpanda committed:
MED-02
+ FIX CRUD operations for patient from Konrad comments second part from classes
